File: ScheduleBox/Model/SchedulesResponse.cs
namespace ScheduleBox.Model
{
    using System;
    using System.Collections.Generic;
    using System.Linq;

    public class SchedulesResponse
    {
        public SchedulesResponse(IReadOnlyList<DaySchedule> schedules)
        {
            this.Schedules = schedules;
            this.Start = schedules.SelectMany(x => x.Activities).Min(x => x.Start);
            this.End = schedules.SelectMany(x => x.Activities).Max(x => x.End);
        }

        public DateTimeOffset Start { get; }

        public DateTimeOffset End { get; }

        public IReadOnlyList<DaySchedule> Schedules { get; }
    }
}

File: ScheduleBox/ClientApp/src/app/schedule/SchedulesResponse.ts
export interface SchedulesResponse {
  start: string;
  end: string;
  schedules: Schedule[];
}

export interface Person {
  name: string;
  id: string;
}

export interface Activity {
  description: string;
  color: string;
  start: string;
  end: string;
}

export interface Schedule {
  person: Person;
  activities: Activity[];
}
